python/GafferUI/ColorSwatch.py

```diff
@@ -126,15 +128,32 @@ def __updateCheckerColors( self ) :
 			self.__transparentChecker.color1 = self._qtColor( effectiveDisplayTransform( color1 ) )
 
 		## \todo Colour should come from the style when we have styles applying to Widgets as well as Gadgets
-		self.__opaqueChecker.borderColor = QtGui.QColor( 119, 156, 255 ) if self.getHighlighted() else None
-		self.__transparentChecker.borderColor = QtGui.QColor( 119, 156, 255 ) if self.getHighlighted() else None
+		if not self.__errored:
+			self.__opaqueChecker.borderColor = QtGui.QColor( 119, 156, 255 ) if self.getHighlighted() else None
+			self.__transparentChecker.borderColor = QtGui.QColor( 119, 156, 255 ) if self.getHighlighted() else None
+		else:
+			self.__opaqueChecker.borderColor = QtGui.QColor( 255, 85, 85 )
+			self.__transparentChecker.borderColor = QtGui.QColor( 255, 85, 85 )
 
 		self._qtWidget().update()
 
 	def __displayTransformChanged( self ) :
 
 		self.__updateCheckerColors()
 
+	def setErrored( self, errored ) :
+
+		if errored == self.getErrored() :
+			return
+
+		self.__errored = errored
+		self._repolish()
```

Comment:
Repolishing is something we do to force Qt to reevaluate the stylesheet, which is not relevant here. I think we need to call `self.__updateCheckerColors()` instead? At the moment, the border isn't updating when an error occurs.

Reply:
I've made this change, but in my test it was working before, so I don't know if it fixed it.

Reply:
It's working nicely for me now...